### PR TITLE
54239 fleet manager stub for ft

### PR DESCRIPTION
**Pull request by sumod13943**
New endpoints added particularly for functional tests with 10 products in the file.

### CODE REVIEW

**Review by Vishal14583 — APPROVED** [review]
LGTM

File: UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs [hunk 1]
```diff
@@ -136,6 +136,119 @@ public IActionResult GetCatalogue([FromHeader(Name = "token")] string? token, [F
...
+            }
+            string? userName = _fleetManagerApiConfiguration.Value.UserName;
+            string? password = _fleetManagerApiConfiguration.Value.Password;
+            string? fleetManagerStubSubscriptionKey = _fleetManagerApiConfiguration.Value.SubscriptionKey;
+            string base64Credentials = GetBase64EncodedCredentials(userName, password);
+            HttpResponseMessage httpResponse = new();
+            string AuthToken = string.Empty;
+            if (string.IsNullOrEmpty(subscriptionKey))
+            {
+                string? invalidSubscriptionKeyResponse = "{\"statusCode\": 401, \"message\":\"Access denied due to missing subscription key. Make sure to include subscription key when making requests to an API.\"}";
```
Comment (ajasonca) [inline UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs:158]: invalidSubscriptionKeyResponse doesn't need to be nullable, and can be made into a const.

File: UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs [hunk 2]
```diff
@@ -136,6 +136,119 @@ public IActionResult GetCatalogue([FromHeader(Name = "token")] string? token, [F
...
+                if (!httpResponse.IsSuccessStatusCode)
+                {
+                    return Unauthorized(JsonConvert.DeserializeObject<InvalidSubscriptionKeyResponse>(invalidSubscriptionKeyResponse));
+                }
+            }
+            if (subscriptionKey == fleetManagerStubSubscriptionKey)
+            {
+                if (userPass == base64Credentials)
+                {
+                    string? authTokenResponse = "{\"token\": \"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9\", \"expiration\":\"" + DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture) + "\"}";
```
Comment (ajasonca) [inline UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs:169]: authTokenResponse doesn't need to be nullable.

File: UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs [hunk 3]
```diff
@@ -136,6 +136,119 @@ public IActionResult GetCatalogue([FromHeader(Name = "token")] string? token, [F
...
+                    httpResponse.StatusCode = HttpStatusCode.BadRequest;
+                    if (!httpResponse.IsSuccessStatusCode)
+                    {
+                        return BadRequest(JsonConvert.DeserializeObject<InvalidUsernamePasswordResponse>(invalidUsernamePasswordResponse));
+                    }
+                }
+            }
+            else
+            {
+                string? invalidSubscriptionKeyResponse = "{\"statusCode\": 401, \"message\":\"Access denied due to invalid subscription key. Make sure to provide a valid key for an active subscription.\"}";
```
Comment (ajasonca) [inline UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs:188]: invalidSubscriptionKeyResponse doesn't need to be nullable, and can be made into a const.

File: UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs [hunk 4]
```diff
@@ -136,6 +136,119 @@ public IActionResult GetCatalogue([FromHeader(Name = "token")] string? token, [F
...
+                    string? authTokenResponse = "{\"token\": \"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9\", \"expiration\":\"" + DateTime.UtcNow.ToString("yyyy-MM-ddTHH:mm:ssZ", CultureInfo.InvariantCulture) + "\"}";
+                    httpResponse.StatusCode = HttpStatusCode.OK;
+                    if (httpResponse.IsSuccessStatusCode)
+                    {
+                        AuthToken = authTokenResponse;
+                    }
+                }
+                else
+                {
+                    string? invalidUsernamePasswordResponse = "{\"correlationId\":\"10ce5b89-67f4-4090-8364-7ba69eb7cb3d\",\"errors\":[{\"source\":\"userPass\",\"description\":\"Missing or invalid encoded User:Pass.\"}]}";
```
Comment (ajasonca) [inline UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs:178]: invalidUsernamePasswordResponse doesn't need to be nullable, and can be made into a const.

File: UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs [hunk 5]
```diff
@@ -136,6 +136,119 @@ public IActionResult GetCatalogue([FromHeader(Name = "token")] string? token, [F
...
+                }
+            }
+            return Ok(JsonConvert.DeserializeObject<JwtAuthUnpToken>(AuthToken));
+        }
+
+        [HttpGet]
+        [Route("/ft/catalogues/{catalogueId}")]
+        public IActionResult GetCatalogueForFT([FromHeader(Name = "token")] string? token, [FromHeader(Name = "Ocp-Apim-Subscription-Key")] string? subscriptionKey)
+        {
+            Dictionary<string, string> requestHeaders = new();
```
Comment (ajasonca) [inline UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs:202]: requestHeaders is not used once declared and populated, so you may as well remove it.

File: UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs [hunk 6]
```diff
@@ -136,6 +136,119 @@ public IActionResult GetCatalogue([FromHeader(Name = "token")] string? token, [F
...
+            {
+                requestHeaders.Add(header.Key, header.Value);
+            }
+            string? fleetManagerStubSubscriptionKey = _fleetManagerApiConfiguration.Value.SubscriptionKey;
+            string? path = _fileDirectoryPathConfiguration.Value.FunctionalTestAVCSCatalogDataFilePath;
+            HttpResponseMessage httpResponse = new();
+
+            if (string.IsNullOrEmpty(subscriptionKey))
+            {
+                string? invalidSubscriptionKeyResponse = "{\"statusCode\": 401, \"message\":\"Access denied due to missing subscription key. Make sure to include subscription key when making requests to an API.\"}";
```
Comment (ajasonca) [inline UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs:213]: invalidSubscriptionKeyResponse doesn't need to be nullable, and can be made into a const.

**Review by ajasonca — CHANGES_REQUESTED** [review]
Some changes to consider - would be worth doing changes to all the methods not just the ones you've changed (e.g. removing the unused variable, change string? to const string, etc.

File: UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs [hunk 7]
```diff
@@ -136,6 +136,119 @@ public IActionResult GetCatalogue([FromHeader(Name = "token")] string? token, [F
             return Unauthorized();
         }
 
+
+        [HttpGet]
+        [Produces("application/json")]
+        [Route("/ft/auth/unp")]
+        public IActionResult GetJwtAuthUnpTokenForFT([FromHeader(Name = "userPass")] string userPass, [FromHeader(Name = "Ocp-Apim-Subscription-Key")] string? subscriptionKey)
+        {
+            Dictionary<string, string> requestHeaders = new();
```
[the remark this answers is not in the document]
Comment (sumod13943) [reply UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs]: Removed

File: UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs (hunk 1 above)
Comment (sumod13943) [reply UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs]: Changed

File: UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs (hunk 2 above)
Comment (sumod13943) [reply UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs]: Changed

File: UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs (hunk 3 above)
Comment (sumod13943) [reply UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs]: Changed

File: UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs (hunk 4 above)
Comment (sumod13943) [reply UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs]: Changed

File: UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs (hunk 5 above)
Comment (sumod13943) [reply UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs]: Changed

File: UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs (hunk 6 above)
Comment (sumod13943) [reply UKHO.FleetManagerMock.API/UKHO.FleetManagerMock.API/Controllers/FleetManagerController.cs]: Changed

**Review by ajasonca — APPROVED** [review]
Looks good

**Pull request merged**